book/src/concepts/type-conversions.md

```diff
@@ -0,0 +1,66 @@
+<!--
+SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+
+SPDX-License-Identifier: MIT OR Apache-2.0
+-->
+# Type Conversions
+
+As the primary use for CXX-Qt is to describe a Qt API in Rust, this means that for technical reasons the required Qt API will sometimes be slightly different from the Rust API.
+
+For example, when dealing with [opaque types](./types.html#custom-opaque-types), these can never be passed by-value from/to Rust.
+In most cases, you'll therefore use a UniquePtr in Rust when you need ownership of such an object.
+
+However, Qt often requires values to be returned by-value, especially when compatibility with QML is required.
+CXX-Qt therefore allows you to automatically convert certain Rust types to certain C++ types.
+
+For example, CXX-Qt can convert a `std::unique_ptr<QColor>` to a `QColor` for you.
+
+CXX-Qt by default provides conversions for:
+
+| From                  | To            |                                           |
```

Comment:
`From C++ type` | `To Rust type`?

Reply:
So for these cases `std::unique_ptr<T>` is the Rust type, and `T` is the C++/Qt type. And it's used for unique_ptr<T> -> T conversions.

But the types don't have to be for "Rust" and "C++", they can be for "C++" and "C++" and could be used elsewhere in our C++ generation or someone elses code.  So they can be generic from T to R as well.